pipeline/transformer/transformer_tei/src/main/scala/weco/pipeline/transformer/tei/transformers/TeiContributors.scala

```diff
@@ -0,0 +1,276 @@
+                          nodeIds: List[String]) = {
+    nodeIds
+      .map(
+        id => addIdToMap(id, scribesMap, c)
+      )
+      .toMap
+  }
+
+  private def addIdToMap(workId: String,
+                         scribesMap: Map[String, List[Contributor[Unminted]]],
+                         c: Contributor[Unminted]) = {
+    scribesMap.get(workId) match {
+      case None =>
+        (workId, List(c))
+      case Some(list) => (workId, list :+ c)
+    }
+  }
+
+  private def extractNestedWorkIds(n: Node) = {
+    (n \ "locus")
+      .map { locus =>
+        (locus \@ "target").trim.replaceAll("#", "").split(" ")
+      }
+      .toList
+      .flatten
+  }
+
+  private def createContributor(
+    label: String,
+    role: ContributionRole): Result[Contributor[Unminted]] =
+    createContributor(label = label, id = "", isFihrist = false, role = role)
+
+  private def createContributor(
+    label: String,
+    id: String,
+    isFihrist: Boolean,
+    role: ContributionRole): Result[Contributor[Unminted]] =
+    (label, id) match {
+      case (l, _) if l.isEmpty =>
+        Left(new RuntimeException(s"The contributor label is empty!"))
+      case (l, id) if id.isEmpty =>
+        Right(Contributor(Person(l), List(role)))
+      case (l, id) =>
+        // If the manuscript is part of the Fihrist catalogue,
+        // the ids of authors refer to the Fihrist authority: https://github.com/fihristorg/fihrist-mss/tree/master/authority
+        // Otherwise they refer to the VIAF authority files: https://viaf.org/
+        val identifierType =
+          if (isFihrist) IdentifierType.Fihrist else IdentifierType.VIAF
+        Right(
+          Contributor(
+            Person(
+              label = l,
+              id = Identifiable(SourceIdentifier(identifierType, "Person", id))
+            ),
+            List(role)
+          )
+        )
+    }
+
+  private def getFromOriginalPersNode(n: Node, list: List[Node]) =
+    list.filter(_ \@ "type" == "original") match {
+      case List(persNode) => getFromPersNode(n, persNode)
+      case Nil =>
+        Left(
+          new RuntimeException(
+            s"No persName nodes with type=original in author $n"
+          )
+        )
+      case _ =>
+        Left(
+          new RuntimeException(
+            s"Multiple persName nodes with type=original in author $n"
+          )
+        )
+    }
+
+  private def getFromPersNode(n: Node, persNode: Node) =
+    Right((persNode.text.trim, getId(n, persNode)))
+
+  private def getFromAuthorNode(n: Node) =
+    Right((n.text.trim, (n \@ "key").trim))
+
+  /**
+    * Sometimes the id of the author is on the persName node
+    * and sometimes it is on the wrapping author node and we must deal with both cases.
+    */
+  private def getId(n: Node, persNode: Node) = {
+    val persNodeId = (persNode \@ "key").trim
+    if (persNodeId.isEmpty) {
+      (n \@ "key").trim
+    } else {
+      persNodeId
+    }
+  }
```

Comment:
Also copied from TeiAuthors, is it a `<pers>` per the variable or a `<persName>` per the comment? Could we make those names consistent?